### ISSUE TITLE
Regarding "Complete" label

### ISSUE THREAD

**Issue by m1xds** [mabeyj/trilium-collection-views#53]
![image](https://github.com/mabeyj/trilium-collection-views/assets/96166421/a529817b-89ef-48ad-9188-71405bf524ab)
Hi maybeyj:

is that possible that by some method we can make this "complete" update automatically? Now it's like "#complete=5" "#total=100", is that possible that we can make it like "#complete=$daycreated" "#total=timeleft"?

**Comment by mabeyj** [comment]
I'm not 100% sure what you're asking. Can you provide more details on how this would work?

Are you asking for Collection Views to have some way of setting certain labels on a note? In this case, do you want it to set `#complete` to `$daycreated` (a creation date of something?) and `#total` to `timeleft` (the current value of a `#timeleft` label?)? For it to do this automatically, when would it do this and how would it know what notes need to be updated?

**Comment by m1xds** [comment]
For example, this note:
![image](https://github.com/mabeyj/trilium-collection-views/assets/96166421/5c8c3fe6-c067-4ce2-aa99-6f1adeabb837)
For example, in this note, the tag "#completed" can only be set to a fixed number, and then the following completed progress bar can be displayed in the total cellection view:
![image](https://github.com/mabeyj/trilium-collection-views/assets/96166421/261fd942-eefb-43cc-9995-3712fe55d79e)
This "progress bar" function is very very very very nice, but now it can only be manually updated every day, I don't know how to make it automatically update, which makes the usefulness greatly reduced, can you make "completed" and total equal to the value in a note, in this example, I just give an example, such as: "#competed=the days differences between the day of creating note with today".
Thank you for your time!

**Comment by mabeyj** [comment]
Makes sense. Currently, this should be possible using Trilium's [daily global event](https://github.com/zadam/trilium/wiki/Events#global-events). A script could run daily that calculates and updates `#completed` and `#total` on each note.

I think it would be feasible to introduce some builtin date expressions like `daysSince($dateCreated)` and `daysBetween($dateCreated,dueDate)` that would allow Collection Views to perform these calculations.

If today is 2024-02-12 and a note was created on 2024-02-09, then `daysSince($dateCreated)` would return 3.  
If `#dueDate` is 2029-02-11, then `daysBetween($dateCreated,dueDate)` would return 1825.

Then combined with a progress bar `#attribute=daysSince($dateCreated),progressBar=daysBetween($dateCreated,dueDate)` would show a bar with "3 / 1,825" today, "4 / 1,825" tomorrow, etc.

**Comment by m1xds** [comment]
Again, thank you so much for your time and detailed explanation!
